conjure-java-core/src/integrationInput/java/com/palantir/product/ListExample.java

```diff
@@ -231,7 +232,14 @@ public Builder primitiveItems(int primitiveItems) {
         @JsonSetter(value = "doubleItems", nulls = Nulls.SKIP, contentNulls = Nulls.FAIL)
         public Builder doubleItems(@Nonnull Iterable<Double> doubleItems) {
             checkNotBuilt();
-            this.doubleItems = ConjureCollections.newArrayList(
+            this.doubleItems = ConjureCollections.newDoubleArrayList(
+                    Preconditions.checkNotNull(doubleItems, "doubleItems cannot be null"));
+            return this;
+        }
+
+        public Builder doubleItems(@Nonnull double[] doubleItems) {
```

Comment:
Adding a new builder methods to allow the setter to create directly from a primitiveArray

Reply:
Let's do one thing at a time -- We should implement this as an optimization without impacting the public api of generated types.